core/heap.c

```diff
@@ -157,7 +159,7 @@ DECLARE_NEVERPROT_VAR(static bool out_of_vmheap_once, false);
 #endif
 
 /* variable-length: we steal one int for the size */
-#define HEADER_SIZE (sizeof(size_t))
+#define HEADER_SIZE HEAP_ALIGNMENT
 /* VARIABLE_SIZE is assignable */
 #define VARIABLE_SIZE(p) (*(size_t *)((p)-HEADER_SIZE))
 #define MEMSET_HEADER(p, value) VARIABLE_SIZE(p) = HEAP_TO_PTR_UINT(value)
```

Comment:
This leaves the 2nd set of 32 bits uninitialized.  As the name indicates, this is supposed to set all the bits in the header, since we do debug scans of memory looking for bit patterns.  I think this has to set the other word to avoid asserts.